tfjs-backend-webgpu/src/binary_op_util.ts

```diff
@@ -202,7 +198,7 @@ const POW_VEC4 = `
   }
   let isNaN = (a < vec4<f32>(0.0)) & (floor(b) < b);
   let valueForNaN = uniforms.NAN;
-  ${CHECK_NAN_SNIPPET_VEC4_INNER}
+  resultTemp = select(resultTemp, vec4<f32>(valueForNaN), isNaN);
```

Comment:
If we're going to inline this, maybe we don't need valueForNaN and even resultTemp.